credoai/modules/dataset_modules/dataset_base.py

```diff
@@ -61,4 +65,86 @@ def _make_pipe(self):
                [('scaler', StandardScaler()),
                 ('model', model)])
         return pipe
+
+    
+    def _find_categorical_features(self, df):
+        """Estimates categorical features
+        Logic: all non-float-type columns are categorical
+
+        Parameters
+        ----------
+        df : pandas.dataframe
+            A dataframe
+
+        Returns
+        -------
+        list
+            Names of categorical features
+        """        
+        cols = df.columns
```

Comment:
I am not comfortable with this automatic procedure.

Someone [online](https://stackoverflow.com/questions/59384802/how-can-i-automatically-detect-if-a-colum-is-categorical) proposed something like this:

```
#category 
for name, column in inputDataFrame.iteritems():
    unique_count = column.unique().shape[0]
    total_count = column.shape[0]
    if unique_count / total_count < 0.05:
        columnTypesDict[name] = 'category'
```

Essentially, they look for columns where the number of unique aspects are a low percentage of the total number. That doesn't work completely on its own either, but seems pretty good for an auto.

Reply:
I came across this too actually, but this then introduces a parameter. How to set that and explain why? 
No program can get it right at the end of the day. I added the considered feature type info to the results for the user to be aware of that. 

```
        results = {}
        for k, v in mi.items():
            if k in categorical_features:
                results[k] = {'value':v, 'feature_type': 'categorical'}
            else:
                results[k] = {'value':v, 'feature_type': 'continuous'}
```

Reply:
I don't think you need to reveal a parameter. It's just our way of identifying categorical variables. Better than just identifying floats I'd think. But your call

Reply:
done